framework/include/utils/MooseEnumBase.h

```diff
@@ -58,14 +62,13 @@ class MooseEnumBase
    * Method for returning a vector of all valid enumeration names for this instance
    * @return a vector of names
    */
-  /// TODO: This should probably be turn into a set to avoid duplicate entries
-  const std::vector<std::string> & getNames() const { return _names; }
+  std::vector<std::string> getNames() const;
```

Comment:
This is because the names must be created on the fly from the MooseEnumItem objects, there is nothing to reference.